### PR TITLE
[feature/dns] metric for timing dns polling sweep

### PR DESCRIPTION
**Pull request by bjlaub**
## After this PR
==COMMIT_MSG==
metric for timing dns polling sweep
==COMMIT_MSG==

### CODE REVIEW

File: dialogue-clients/src/main/java/com/palantir/dialogue/clients/DialogueDnsResolutionWorker.java [hunk 1]
```diff
@@ -64,7 +69,7 @@ public void run() {
                 // this logging may be helpful in informing us of problems in the system.
                 log.info("Output refreshable has been garbage collected, no need to continue polling");
             } else {
-                doUpdate(null);
+                updateTimer.time(() -> doUpdate(null));
```
Comment (bjlaub) [inline dialogue-clients/src/main/java/com/palantir/dialogue/clients/DialogueDnsResolutionWorker.java]: whoops, this is the wrong place to put this, i think we need to measure within doUpdate.

File: dialogue-clients/src/main/java/com/palantir/dialogue/clients/DialogueDnsResolutionWorker.java [hunk 2]
```diff
@@ -104,5 +111,7 @@ private synchronized void doUpdate(@Nullable INPUT updatedInputState) {
                 log.info("Attempted to update DNS output refreshable which has already been garbage collected");
             }
         }
+        long end = System.nanoTime();
+        updateTimer.update(Duration.ofNanos(end - start));
```
Comment (carterkozak) [inline dialogue-clients/src/main/java/com/palantir/dialogue/clients/DialogueDnsResolutionWorker.java]: We may want to put this inside the `if (inputState != null) {` conditional to make sure we're actually measuring work

File: dialogue-clients/src/main/metrics/dialogue-core-metrics.yml [hunk 3]
```diff
@@ -11,3 +11,6 @@ namespaces:
           - name: kind
             docs: Describes the type of component polling for DNS updates.
         docs: Number of active Dialogue DNS update background tasks currently scheduled.
+      resolveTime:
+        type: timer
```
Comment (carterkozak) [inline dialogue-clients/src/main/metrics/dialogue-core-metrics.yml:15]: Can we tag this with the `kind` tag used above, from the `DnsPollingSpec`?

Comment (bjlaub) [reply dialogue-clients/src/main/metrics/dialogue-core-metrics.yml:15]: yep will do

File: dialogue-clients/src/main/java/com/palantir/dialogue/clients/DialogueDnsResolutionWorker.java [hunk 4]
```diff
@@ -103,6 +109,8 @@ private synchronized void doUpdate(@Nullable INPUT updatedInputState) {
             } else {
                 log.info("Attempted to update DNS output refreshable which has already been garbage collected");
             }
+            long end = System.nanoTime();
+            updateTimer.update(Duration.ofNanos(end - start));
```
Comment (carterkozak) [inline dialogue-clients/src/main/java/com/palantir/dialogue/clients/DialogueDnsResolutionWorker.java]: nit: let's avoid Duration allocation and use `updateTimer.update(end - start, TimeUnit.NANOSECONDS)`

File: dialogue-clients/metrics.md [hunk 5]
```diff
@@ -55,6 +55,8 @@ Connection pool metrics from the dialogue Apache client.
 Dialogue DNS metrics.
 - `client.dns.tasks` (counter): Number of active Dialogue DNS update background tasks currently scheduled.
   - `kind`: Describes the type of component polling for DNS updates.
+- `client.dns.resolveTime` (timer): Measures the time taken to complete a full pass polling for DNS updates.
```
Comment (carterkozak) [inline dialogue-clients/metrics.md]: perhaps something like `client.dns.refresh` to make the name reflect that this is a full sweep

Comment (bjlaub) [reply dialogue-clients/metrics.md]: done

File: dialogue-clients/src/main/java/com/palantir/dialogue/clients/DialogueDnsResolutionWorker.java (hunk 4 above)
Comment (bjlaub) [reply dialogue-clients/src/main/java/com/palantir/dialogue/clients/DialogueDnsResolutionWorker.java]: fixed

**Review by carterkozak — APPROVED** [review]

**Pull request merged**